src/terminal/parser/InputStateMachineEngine.cpp

```diff
@@ -454,6 +462,29 @@ bool InputStateMachineEngine::ActionCsiDispatch(const VTID id, const VTParameter
     case CsiActionCodes::FocusOut:
         success = _pDispatch->FocusChanged(false);
         break;
+    case CsiActionCodes::DA_DeviceAttributes:
+        // This assumes that InputStateMachineEngine is tightly coupled with VtInputThread and the rest of the ConPTY system.
+        // On startup, ConPTY will send a DA1 request to get more information about the hosting terminal.
+        // We catch it here and store the information for later retrieval.
+        if (_deviceAttributes.load(std::memory_order_relaxed) == 0)

```

Comment:
if we get this while we're `_lookingForDSR`, but we don't get a DSR, should we disable looking for the cursor at that point?

Reply:
Oh! I forgot about that!
Thanks!